src/model.rs

```diff
@@ -2840,5 +2861,37 @@ mod tests {
             assert_eq!(3, stop_time.pickup_type);
             assert_eq!(3, stop_time.drop_off_type);
         }
+
+        #[test]
+        fn remove_pickup_drop_off_type_3() {
+            let model = transit_model_builder::ModelBuilder::default()
+                .vj("vj1", |vj| {
+                    vj.st_mut("SP1", "10:00:00", "10:01:00", |st| {
+                        st.pickup_type = 0;
+                        st.drop_off_type = 1;
+                    })
+                    .st_mut("SP2", "11:00:00", "11:01:00", |st| {
+                        st.pickup_type = 3;
+                        st.drop_off_type = 2;
+                    })
+                    .st_mut("SP3", "12:00:00", "12:01:00", |st| {
+                        st.pickup_type = 3;
+                        st.drop_off_type = 2;
+                    })
+                    .st_mut("SP4", "13:00:00", "13:01:00", |st| {
+                        st.pickup_type = 1;
+                        st.drop_off_type = 0;
+                    });
+                })
+                .build();
+
+            let mut collections = model.into_collections();
+            collections.remove_points_of_route();
+            let vj = collections
+                .vehicle_journeys
+                .get("vj1")
+                .expect("Failed to find vehicle journey vj1");
```

Comment:
minor: we actually don't expect to "Failed to find vehicle journey vj1", we actually expect to **find** it (but all the other tests are doing the same weirdness, so I'm fine if this is not fixed here).